packages/protocol/contracts/governance/VestingInstance.sol

```diff
@@ -0,0 +1,356 @@
+pragma solidity ^0.5.3;
+
+import "openzeppelin-solidity/contracts/math/SafeMath.sol";
+import "../common/UsingRegistry.sol";
+
+contract VestingSchedule is UsingRegistry {
+
+    modifier onlyRevoker() {
+      require(msg.sender == mRevoker, "sender must be the vesting revoker");
+      _;
+    }
+
+    modifier onlyBeneficiary() {
+      require(msg.sender == mRevoker, "sender must be the vesting beneficiary");
+      _;
+    }
+
+    using SafeMath for uint256;
+
+    event VestingGoldLocked(uint256 amount, uint256 timestamp);
+    event VestingGoldUnlocked(uint256 amount, uint256 timestamp);
+    event VestingGoldRelocked(uint256 index, uint256 timestamp);
+    event VestingGoldWithdrawn(uint256 index, uint256 timestamp);
+    event VestingAccountVoterAuthorized(address authorizer, address voter, uint256 timestamp);
+    event VestingAccountValidatorAuthorized(address authorizer, address validator, uint256 timestamp);
+    event VestingAccountAttestatorAuthorized(address authorizer, address attestator, uint256 timestamp);
+    event VestingWithdrawn(address beneficiary, uint256 amount, uint256 timestamp);
+    event VestingRevoked(address revoker, address refundDestination, uint256 refundDestinationAmount, uint256 timestamp);
+
+    // total that is to be vested
+    uint256 public mVestingAmount;
```

Comment:
Do we need this if we have the amount per period and the number of periods?

Reply:
as discussed today (pls. see ticket specs)